src/steam/ISteamClient.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Godot;
+
+/// <summary>
+///   Interface specifying the Steam operations we use. Implemented in the SteamClient.cs file
+/// </summary>
+public interface ISteamClient : ISteamSignalReceiver
+{
+    /// <summary>
+    ///   True if steam client has been initialized
+    /// </summary>
+    bool IsLoaded { get; }
+
+    string DisplayName { get; }
+
+    void Init();
+
+    /// <summary>
+    ///   Sets up receiver to receive callbacks
+    /// </summary>
+    /// <param name="receiver">The receiver. Must implement <see cref="ISteamSignalReceiver"/></param>
+    /// <remarks>
+    ///   <para>
+    ///     Note that for many of the received signals they need to be forwarded to this object for further processing.
+    ///   </para>
+    /// </remarks>
+    void ConnectSignals(Object receiver);
```

Comment:
Wouldn't something like this work here?
```cs
    void ConnectSignals<T>(T receiver)
        where T : Object, ISteamSignalReceiver;
```

Reply:
That does indeed work.

Reply:
https://github.com/Revolutionary-Games/Thrive/pull/2720/commits/4bfc47350eed44a4673e7fd42c17e0d38c122bed